src/engine/Match_extract_mode.ml

```diff
@@ -51,6 +51,10 @@ type match_result_location_adjuster =
   Core_profiling.partial_profiling Core_result.match_result ->
   Core_profiling.partial_profiling Core_result.match_result
 
+(* We need to keep track of the original target for each extracted one
+   for things like counting the number of scanned original files *)
+type assoc_extracted_target_to_original = string * Fpath.t
```

Comment:
add Hashtbl.t in the type alias, and then use it in Core_scan.mli and other files.

Reply:
If I'm following correctly, I suggest using a record (later) in which a field is
```
assoc_extracted_target_to_original : string * Fpath.t;
```
I don't like aliases much because they don't appear in error messages or worse, the wrong alias is used by the type checker in error messages.